File: 33-day/pitagoras.py
def quadraticEquationSolutions(A, B, C):
    if A == 0:
        if B == 0:
            if C == 0:
                return 'There are infinite solutions'
            else:
                return 'There are no solutions'
        else:
            return 'x = {}'.format(-C/B)
    else:
        d = (B ** 2 - 4 * A * C) **.5
        return 'x1 = {}, x2 = {}'.format((-B + d) / (2 * A), (-B - D) / (2 * A))


# x^2 - 9x + 20 = 0
print(quadraticEquationSolutions(1, -9, 20))
# x1 = 5.0, x2 = 4.0

# x^2 - 4x + 5 = 0
print(quadraticEquationSolutions(1, -4, 5))
# x1 = (2+1j), x2 = (2-1j)

# 5x - 15 = 0
print(quadraticEquationSolutions(0, 5, -10))
# x = 2.0

# 8 = 0
print(quadraticEquationSolutions(0, 0, 8))
# There are no solutions

# 0 = 0
print(quadraticEquationSolutions(0, 0, 0))
# There are infinite solutions

File: 33-day/pitagoras.js
function quadraticEquationSolutions(A, B, C) {
    let solution = { x1: 0, x2: 0 }
    if (A === 0) {
        if (B === 0) {
            return C === 0
                ? 'There are infinite solutions'
                : 'There are no solutions'
        } else {
            return { x: -C / B }
        }
    } else {
        let dSquare = B ** 2 - 4 * A * C
        if (dSquare < 0) {
            return 'The solutions are imaginary'
        } else {
            let d = Math.sqrt(dSquare)
            return { x1: (-B + d) / (2 * A), x2: (-B - d) / (2 * A) }
        }
    }
}


// x^2 - 9x + 20 = 0
console.log(quadraticEquationSolutions(1, -9, 20))
// {x1: 5, x2: 4}

// x^2 - 4x + 5 = 0
console.log(quadraticEquationSolutions(1, -4, 5))
// The solutions are imaginary

// 5x - 15 = 0
console.log(quadraticEquationSolutions(0, 5, -10))
// {x: 2}

// 8 = 0
console.log(quadraticEquationSolutions(0, 0, 8))
// There are no solutions

// 0 = 0
console.log(quadraticEquationSolutions(0, 0, 0))
// There are infinite solutions